tracks/rust/exercises/atbash-cipher/mentoring.md

```diff
@@ -0,0 +1,164 @@
+# Atbash Cipher
+
+## Concepts
+
+- ascii
+- chars
+- iterators
+- primative types
+
+## Reasonable solutions
+
+A reasonable solution should do the following:
+
+- Use byte literals rather than integer literals to represent ascii characters.
+  Use `b'a'` rather than `97`.
+- Use char methods for working with ascii data, all of which contain `_ascii_`
+  in their name, rather than filtering characters with `is_ascii` and using
+  unicode centric methods for manipulation.
+- Factor the common logic into a seperate function rather than duplicating
+  filtering and transposition in encode and decode. **Bonus** its fine if this
+  function returns a String or if they just use decode as the starting point of
+  encode, but this is good place to introduce the `impl Trait` in the return
+  position after having already approved the exercise.
+- **Optional**: Avoid heap allocations when inserting spaces into the encoded
+  final string. Its better to `flat_map` spaces into the than collecting each
+  set of chunks and joining the slices together with spaces. Using itertools
+  `chunks` and `join` is okay but can cause lifetime issues that are fixed with
+  NLL, its okay for students so use this but if they do its a good idea to show
+  them alternative solutions that use only std methods.
+
+### Examples
+
+```rust
+use std::iter::once;
+
+static ALPHABET: &'static str = "abcdefghijklmnopqrstuvwxyz";
+
+/// "Encipher" with the Atbash cipher.
+pub fn encode(plain: &str) -> String {
+    atbash(plain)
+        .enumerate()
+        .flat_map(|(ind, c)| {
+            match ind % 5 {
```

Comment:
I do not have the exact algorithm in my mind, but `5` feels quite like a magic number.

Reply:
I believe 5 is the group size you're supposed to split the output into.

Reply:
Usually I'd ask students to use a constant in such cases.

Reply:
I'll add it as a possible comment. But I would tend to see that as a style issue rather than a rust issue and I tend to be hesitant about giving those kinds of comments for fear of push back.